client/src/pages/locations/GeoLocation.js

```diff
@@ -139,66 +131,67 @@ const MGRSFormField = ({
           <Field
             name="displayedCoordinate"
             component={FieldHelper.InputFieldNoLabel}
-            onChange={e => updateFields(e.target.value)}
+            onChange={e => updateCoordinatesOnChange(e.target.value)}
             onBlur={e => {
-              updateFields(e.target.value)
+              updateCoordinatesOnBlur(e.target.value)
             }}
           />
         </Col>
         <CoordinateActionButtons
-          lat={lat}
-          lng={lng}
+          coordinates={coordinates}
           isSubmitting={isSubmitting}
-          disabled={!values.displayedCoordinate}
+          disabled={!coordinates.displayedCoordinate}
           onClear={() => {
-            setFieldValue("displayedCoordinate", null)
+            setFieldTouched("displayedCoordinate", false, false)
+            setFieldValue("displayedCoordinate", null, false)
             setFieldValue("lat", null, false)
             setFieldValue("lng", null, false)
           }}
         />
       </Col>
     </FormGroup>
   )
-
-  function updateFields(val) {
-    setFieldTouched("displayedCoordinate", true, false)
-    const newLatLng = convertMGRSToLatLng(val)
+  // Lat-Lng fields are read only, no need to validate in onChange or onBlur
+  function updateCoordinatesOnChange(val) {
     setFieldValue("displayedCoordinate", val)
+    const newLatLng = convertMGRSToLatLng(val)
+    setFieldValue("lat", newLatLng[0], false)
+    setFieldValue("lng", newLatLng[1], false)
+  }
+  function updateCoordinatesOnBlur(val) {
+    setFieldTouched("displayedCoordinate", true)
+    const newLatLng = convertMGRSToLatLng(val)
     setFieldValue("lat", newLatLng[0], false)
     setFieldValue("lng", newLatLng[1], false)
   }
 }
 
 MGRSFormField.propTypes = {
-  lat: PropTypes.oneOfType([PropTypes.number, PropTypes.string]),
-  lng: PropTypes.oneOfType([PropTypes.number, PropTypes.string]),
+  coordinates: PropTypes.object,
   editable: PropTypes.bool,
   setFieldValue: fnRequiredWhenEditable,
   setFieldTouched: fnRequiredWhenEditable,
-  isSubmitting: PropTypes.bool,
-  values: PropTypes.object
+  isSubmitting: PropTypes.bool
 }
 
 MGRSFormField.defaultProps = {
-  lat: null,
-  lng: null,
+  coordinates: DEFAULT_COORDINATES,
   editable: false,
   isSubmitting: false
 }
 
 /* ========================= LatLonFormField ================================ */
 
 const LatLonFormField = ({
-  lat,
-  lng,
+  coordinates,
   editable,
   setFieldValue,
   setFieldTouched,
   isSubmitting
 }) => {
   if (!editable) {
-    const lt = parseCoordinate(lat)
-    const ln = parseCoordinate(lng)
+    const lt = coordinates.lat
```

Comment:
I would name the new variables lat and lng, otherwise it is harder to read the code. `lt` and `ln` doesn't directly have a meaning for me, unless I know the context.

Reply:
`const { lat, lng } = coordinates`